### PR TITLE
feat: improve headers config and types

### PR DESCRIPTION
**Pull request by Qrzy**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This change aims to:
- update the misleading `| boolean` in types where `true` is actually an unacceptable value that would break the module
- provide better typing for certain headers' values
- enable modifying only picked parts of complex headers like Content Security Policy

Especially the very last point pushed me towards proposing this change as I found it really cumbersome to copy, modify and paste the original value for CSP when trying to add [Nuxt Image](https://v1.image.nuxtjs.org/get-started/) using Cloudinary.

Change should be fully backwards-compatible (see using the string value when it's given instead of an object) - if there is any breaking change - I'll definitely rethink it and try to improve this proposal.

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why) - well, there are no test in the repo 🙃

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -26,40 +26,46 @@ export default defineNuxtModule<ModuleOptions>({
...
-          nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route] = { headers: nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route] ? { ...nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route].headers } : {} }
-
-          nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route].headers!![SECURITY_HEADER_NAMES[header]] = (nuxt.options.security.headers as SecurityHeader)[header].value
+    if (securityOptions.headers) {
+      for (const header in securityOptions.headers as SecurityHeaders) {
+        const headerOptions = (securityOptions.headers as SecurityHeader)[header]
+        if (headerOptions) {
+          const nitroRouteRules = nuxt.options.nitro.routeRules
+          nitroRouteRules!![headerOptions.route] = {
+            ...nitroRouteRules!![headerOptions.route],
```
Comment (Qrzy) [inline src/module.ts:50]: Also, added this little fella, as I found the previous version disregarding any potential previous of entire route rules.

**Comment by Baroshem** [comment]
@Qrzy, I think that I have rebased the code in a wrong way (as it was a lot of files to adjust). Could you please revert it so that I can adjust it to the current main?

Sorry for the trouble.

**Comment by Qrzy** [comment]
@Baroshem, this should be ready for review now :)

**Comment by Baroshem** [comment]
Thanks @Qrzy !

I will merge it today and begin testing!

**Pull request merged**